noxfile.py

```diff
@@ -0,0 +1,70 @@
+import nox
+import os
+import sys
+
+
+@nox.session(name="pybamm-requires")
+def run_pybamm_requires(session):
+    if sys.platform != "win32" or sys.platform != "darwin":
+        session.install("wget", "cmake")
+        session.run("python", "scripts/install_KLU_Sundials.py")
+        session.run(
+            "git",
+            "clone",
+            "https://github.com/pybind/pybind11.git",
+            "pybind11/",
+            external=True,
+        )
+
+
+@nox.session(name="coverage")
+def run_coverage(session):
+    homedir = os.getenv("HOME")
+    session.env["SUNDIALS_INST"] = session.env.get("SUNDIALS_INST", f"{homedir}/.local")
+    session.env[
+        "LD_LIBRARY_PATH"
+    ] = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    session.install("coverage", "scikits.odes")
```

Comment:
This should ideally go in the if statement below. We don't really want to install `scikits.odes` on every OS. Installing `coverage` here should be fine.

Reply:
Now `scikits.odes` would only be installed in GNU/Linux.

Reply:
I don't think this should be the case, since `scikits.odes` does get installed on macOS when one runs `tox -e coverage`. The difference is that on Linux, we should use the `pybamm_install_odes` command, and on macOS, we should use `pip install scikits.odes`

Reply:
`tox -e coverage` never runs on MacOS, so we currently only test the solver on Linux. `pybamm_install_odes` is not really used anywhere in the CI, the script is only meant for users (see #2962).

`pybamm_install_odes` calls `pip install scikits.odes` internally, and acts almost identical to `tox -e coverage`, the only difference being that it does not build `SUNDIALS` with `KLU`. We can use `pip install scikits.odes` on any OS, given that we have `SUNDIALS` installed correctly.

Reply:
Thanks, I get it now – we shouldn't use `pybamm_install_odes` in the CI that way. However, though `tox -e coverage` does not run on macOS in the CI (it runs only for Linux), it is intended to be run locally too on all three platforms. So I suggest we install `scikits.odes` on macOS too with `nox -s coverage`, just to replicate the functionality of the local macOS `tox -e coverage`. 

Otherwise, we should have different `coverage` environments for CI and for running locally (which does not seem to be a good idea, obviously)